app/shell-window/ui/navbar.js

```diff
@@ -323,17 +324,18 @@ function decorateResultMatches (searchTerms, result) {
     // use the length of the search term
     // (sqlite FTS gives the length of the full matching token, which isnt as helpful)
     let searchTerm = searchTerms[termIndex]
-    let len = searchTerm.length
+    // offset from sqlite is based on bytes. we need to calculate the length of searchTerm in bytes too.
+    let len = byteLength(searchTerm)
```

Comment:
since the offset returned from SQLite is based on bytes. this function will decorate wrong segments since it's splitting segments based on character length. Here is an example(highlighted with red so it's easier to see):

![2016-09-22 12 08 00](https://cloud.githubusercontent.com/assets/8631/18719707/11594492-805b-11e6-9065-97cd51824133.png)

This commit fixed the problem by splitting segments based on bytes.